### ISSUE TITLE
Snapshot guid changed to string on migration removal

### ISSUE THREAD

**Issue by IvMisticos** [PomeloFoundation/Pomelo.EntityFrameworkCore.MySql#1231]
### Steps to reproduce
1. Create a database with entities that contain a field that is a Guid
2. Create a migration
3. Remove a migration
4. Guids became strings

### The issue
Initially after creating a migration Guids look like that:
```cs
b.Property<Guid>("MyGuid")
```

After removing the migration:
```cs
b.Property<string>("MyGuid")
```

```cs
Exception message: No mapping to a relational type can be found for property 'MyGuid' with the CLR type 'string'.
```

### Further technical details
MySQL version: MariaDB 10.4
Pomelo.EntityFrameworkCore.MySql version: 3.2.3
Target Framework: .NET Core 3.1
Microsoft.EntityFrameworkCore: 3.1.9

**Comment by lauxjpn** [comment]
This is a duplicate of #1002 (and also #1212). We reported the issue to the EF Core team with dotnet/efcore#19637.
According to the EF Core reported issue, it is possible that this issue is resolved in 5.0. We have not checked that yet, but our first official `5.0.0-alpha.1` release is out since a couple of hours, so feel free to test it in this regard and report back to us on your results in the #1002 issue.

**Comment by IvMisticos** [comment]
Thanks, not sure how I have missed that - was searching for something else so might have forgotten.
